test/t8_schemes/t8_gtest_face_neigh.cxx

```diff
@@ -0,0 +1,251 @@
+
+  for (int iface = 0; iface < num_faces; iface++) {
+    /* Compute the neighbors neighbor along a given face and check, if the result is the
+     * original element. */
+    ts->t8_element_face_neighbor_inside (child, neigh, iface, &face_num);
+    ts->t8_element_face_neighbor_inside (neigh, element, face_num, &check);
+
+    EXPECT_EQ (ts->t8_element_compare (child, element),
+               0) << "Got a false neighbor.";
+    EXPECT_EQ (check, iface) << "Wrong face.";
+  }
+}
+
+int
+t8_test_get_middle_child (t8_eclass_t eclass, int ilevel,
+                          t8_element_t *element, t8_element_t *child,
+                          t8_eclass_scheme_c *ts)
+{
+  /* Get the child number of the child in the middle of the element, depending of the shape of the element. */
+  switch (eclass) {
+  case T8_ECLASS_VERTEX:
+    return 0;
+  case T8_ECLASS_LINE:
+    return 0;
+  case T8_ECLASS_QUAD:
+    /* There are no inner children in level one refinement. The test starts with level two, because this is the first level, inner children exists.
+       The third child of level one child 0 is one of four middle children in level two. */
+    ts->t8_element_child (element, 0, child);
+    ts->t8_element_copy (child, element);
+    return 3;
+  case T8_ECLASS_TRIANGLE:
+    return 3;
+  case T8_ECLASS_HEX:
+    /* There are no inner children in level one refinement. The test starts with level two, because this is the first level, inner children existing.
+       The third child of level one child 4 is one of eight middle children in level two. */
+    ts->t8_element_child (element, 4, child);
+    ts->t8_element_copy (child, element);
+    return 3;
+  case T8_ECLASS_TET:
+    return 3;
+  case T8_ECLASS_PRISM:
+    /* There are no inner children in level one refinement. The test starts with level two, because this is the first level, inner children existing.
+       The last child of level one child 4 is one of eight middle children in level two. */
+    ts->t8_element_child (element, 4, child);
+    ts->t8_element_copy (child, element);
+    return 7;
+  case T8_ECLASS_PYRAMID:
+    {
+      t8_dpyramid_t      *pyramid = (t8_dpyramid_t *) element;
+      /* middle_child_id of Pyramid Type 6. */
+      if (pyramid->pyramid.type == T8_DPYRAMID_FIRST_TYPE) {
+        return 8;
+      }
+      /* middle_child_id of Pyramid Type 7. */
+      else {
+        return 3;
+      }
+    }
+  default:
+    return 0;
+  }
+}
+
+/* First "simple" check. First, the neighbors of the root-element at level 0 are computed
+ * which should all lie outside. Then, the child is constructed and it is checked,
+ * if if all neighbors are computed correctly. Then, the same is done for all of the children. 
+ * The same is done until level maxlvl. */
+TEST_P (face_neigh, face_check_easy)
+{
+  int                 middle_child_id;
+
+  /* Are the neighbors of the element really outside?. */
+  t8_check_not_inside_root (element, neigh, child, ts);
+
```

Comment:
I suggest moving this into its own `TEST_P`, as it is completely independent of the rest of the checks in this test.